bancor_research/bancor_simulator/v3/spec/utils.py

```diff
@@ -53,32 +53,33 @@ def compute_user_total_holdings(state: State, user_name: str):
     return sum(holdings)
 
 
-def compute_pooltoken_amt(state: State, tkn_name: str, tkn_amt: Decimal) -> Decimal:
+def compute_rtkn_amt(state: State, tkn_name: str, ptkn_amt: Decimal) -> Decimal:
     """
-    Returns the pool_token_amt for a given tkn_name, tkn_amt.
+    Returns the reserve token amount for a given pool token amount.
     """
-    staked_amt = get_staked_balance(state, tkn_name)
+    staked_balance = get_staked_balance(state, tkn_name)
     pool_token_supply = get_pooltoken_balance(state, tkn_name)
-    if staked_amt > 0:
-        pool_token_amt = (lambda a, b, c: a * b / c)(
-            pool_token_supply, tkn_amt, staked_amt
-        )
-    else:
-        pool_token_amt = Decimal("0")
-    return pool_token_amt
+    return ptkn_amt * staked_balance / pool_token_supply
+
+
+def compute_ptkn_amt(state: State, tkn_name: str, rtkn_amt: Decimal) -> Decimal:
+    """
+    Returns the pool token amount for a given reserve token amount.
+    """
+    staked_balance = get_staked_balance(state, tkn_name)
+    pool_token_supply = get_pooltoken_balance(state, tkn_name)
+    return rtkn_amt * pool_token_supply / staked_balance
 
```

Comment:
These changes (above) require a careful review.

They include:
- Adding function `compute_rtkn_amt`
- Renaming function `compute_pooltoken_amt` to `compute_ptkn_amt`

I believe that we should let each one of these functions to revert on division by zero when the staked amount happens to be zero.

Reply:
Im good with this approach.

Reply:
Also this looks correct but I have not tested it, have you?